frame/bounties/src/lib.rs

```diff
@@ -0,0 +1,827 @@
+//! - **Curator fee:** The reserved upfront payment for a curator for work related to the bounty.
+//!
+//! ## Interface
+//!
+//! ### Dispatchable Functions
+//!
+//! General spending/proposal protocol:
+//! - `propose_spend` - Make a spending proposal and stake the required deposit.
+//! - `reject_proposal` - Reject a proposal, slashing the deposit.
+//! - `approve_proposal` - Accept the proposal, returning the deposit.
+//!
+//! Tipping protocol:
+//! - `report_awesome` - Report something worthy of a tip and register for a finders fee.
+//! - `retract_tip` - Retract a previous (finders fee registered) report.
+//! - `tip_new` - Report an item worthy of a tip and declare a specific amount to tip.
+//! - `tip` - Declare or redeclare an amount to tip for a particular reason.
+//! - `close_tip` - Close and pay out a tip.
+//!
+//! Bounty protocol:
+//! - `propose_bounty` - Propose a specific treasury amount to be earmarked for a predefined set of
+//! tasks and stake the required deposit.
+//! - `approve_bounty` - Accept a specific treasury amount to be earmarked for a predefined body of work.
+//! - `propose_curator` - Assign an account to a bounty as candidate curator.
+//! - `accept_curator` - Accept a bounty assignment from the Council, setting a curator deposit.
+//! - `extend_bounty_expiry` - Extend the expiry block number of the bounty and stay active.
+//! - `award_bounty` - Close and pay out the specified amount for the completed work.
+//! - `claim_bounty` - Claim a specific bounty amount from the Payout Address.
+//! - `unassign_curator` - Unassign an accepted curator from a specific earmark.
+//! - `close_bounty` - Cancel the earmark for a specific treasury amount and close the bounty.
+//!
+//!
+//! ## GenesisConfig
+//!
+//! The Treasury module depends on the [`GenesisConfig`](./struct.GenesisConfig.html).
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+mod tests;
+mod benchmarking;
+pub mod weights;
+
+use sp_std::prelude::*;
+
+use sp_std::if_std;
+
+use frame_support::{decl_module, decl_storage, decl_event, ensure, print, decl_error};
+
+use frame_support::traits::{
+	Currency, Get, Imbalance, OnUnbalanced, ExistenceRequirement::{KeepAlive, AllowDeath},
+	ReservableCurrency, WithdrawReasons
+};
+use sp_runtime::{Permill, ModuleId, RuntimeDebug, DispatchResult, traits::{
+	Zero, StaticLookup, AccountIdConversion, Saturating, BadOrigin
+}};
+use frame_support::dispatch::DispatchResultWithPostInfo;
+use frame_support::traits::{Contains, ContainsLengthBound, EnsureOrigin};
+use frame_support::weights::{Weight, DispatchClass};
+
+use codec::{Encode, Decode};
+use frame_system::{self as system, ensure_signed};
+pub use weights::WeightInfo;
+
+type BalanceOf<T, I> = pallet_treasury::BalanceOf<T, I>;
+
+type PositiveImbalanceOf<T, I> = pallet_treasury::PositiveImbalanceOf<T, I>;
+
+pub trait Trait<I=DefaultInstance>: frame_system::Trait + pallet_treasury::Trait<I> {
```

Comment:
In decl_* macros, `I` is used to give the prefix to use for the module, thus Instance1,.. Instance16 is defined in each modules.
here the instance used for bounties pallet can't be the same as the instance used for treasury pallet.

if you use construct_runtime the tests of pallet bounties wouldn't compile.